tests/sentry/api/endpoints/test_organization_sentry_functions.py

```diff
@@ -1,81 +1,137 @@
 from __future__ import annotations
 
-from typing import Any
 from unittest.mock import patch
+from uuid import uuid4
 
-from django.urls import reverse
-
+from sentry.models.sentryfunction import SentryFunction
 from sentry.testutils.cases import APITestCase
-from sentry.testutils.helpers import Feature
+from sentry.testutils.helpers.features import with_feature
 from sentry.testutils.silo import region_silo_test
 
 
-@region_silo_test(stable=True)
-class OrganizationSentryFunctions(APITestCase):
+class OrganizationSentryFunctionBase(APITestCase):
     endpoint = "sentry-api-0-organization-sentry-functions"
 
     def setUp(self):
         super().setUp()
-        self.create_organization(owner=self.user, name="RowdyTiger")
-        self.url = reverse(self.endpoint, args=[self.organization.slug])
         self.login_as(user=self.user)
+        self.code = "exports.yourFunction = (req, res) => {\n\tlet message = req.query.message || req.body.message || 'Hello World!';\n\tconsole.log('Query: ' + req.query);\n\tconsole.log('Body: ' + req.body);\n\tres.status(200).send(message);\n};"
+        self.data = {
+            "name": "foo",
+            "author": "bar",
+            "code": self.code,
+            "overview": "qux",
+        }
+
+
+@region_silo_test(stable=True)
+class OrganizationSentryFunctionsPost(OrganizationSentryFunctionBase):
+    method = "POST"
+
+    def setUp(self):
+        super().setUp()
+        self.login_as(user=self.user)
+        self.data["env_variables"] = [{"name": "foo", "value": "bar"}]
 
+    @with_feature("organizations:sentry-functions")
     @patch("sentry.api.endpoints.organization_sentry_function.create_function")
     def test_post_feature_true(self, mock_func):
-        defaultCode = "exports.yourFunction = (req, res) => {\n\tlet message = req.query.message || req.body.message || 'Hello World!';\n\tconsole.log('Query: ' + req.query);\n\tconsole.log('Body: ' + req.body);\n\tres.status(200).send(message);\n};"
-        data = {
+        response = self.get_success_response(self.organization.slug, status_code=201, **self.data)
+
+        assert response.data == {
             "name": "foo",
+            "slug": "foo",
             "author": "bar",
-            "code": defaultCode,
+            "code": self.code,
             "overview": "qux",
-            "envVariables": [{"name": "foo", "value": "bar"}],
```

Comment:
This is an example of the variable name being incorrect, it should match the serializer:

https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/organization_sentry_function.py#L28